api_app/services/airlock.py

```diff
@@ -91,3 +114,319 @@ def get_airlock_request_container_sas_token(account_name: str,
+
+    user_resource_create = UserResourceInCreate(
+        templateName=user_resource_template_name,
+        properties={
+            "display_name": "Airlock Review VM",
+            "description": f"{airlock_request.title} (ID {airlock_request.id})",
+            "airlock_request_sas_url": airlock_request_sas_url
+        }
+    )
+
+    logging.info(f"Creating a user resource in {workspace.id} {workspace_service_id} {user_resource_template_name}")
+    user_resource, resource_template = await user_resource_repo.create_user_resource_item(
+        user_resource_create, workspace.id, workspace_service_id, workspace_service.templateName, user.id, user.roles)
+
+    operation = await save_and_deploy_resource(
+        resource=user_resource,
+        resource_repo=user_resource_repo,
+        operations_repo=operation_repo,
+        resource_template_repo=resource_template_repo,
+        resource_history_repo=resource_history_repo,
+        user=user,
+        resource_template=resource_template)
+
+    return user_resource, operation
+
+
+async def _handle_existing_review_resource(existing_resource: AirlockReviewUserResource, user: User, user_resource_repo: UserResourceRepository, workspace_service_repo: WorkspaceServiceRepository,
+                                           operation_repo: OperationRepository, resource_template_repo: ResourceTemplateRepository, resource_history_repo: ResourceHistoryRepository):
+    # Is the existing resource enabled, deployed, and can we get its power state information
+    if existing_resource.isEnabled and existing_resource.deploymentStatus == "deployed" and 'azure_resource_id' in existing_resource.properties:
+        resource_status = get_azure_resource_status(existing_resource.properties['azure_resource_id'])
+        if "powerState" in resource_status and resource_status["powerState"] == "VM running":
+            logging.info("Existing review resource is enabled, in a succeeded state and running. Returning a conflict error.")
+            raise HTTPException(status_code=status.HTTP_409_CONFLICT,
+                                detail="A healthy review resource is already deployed for the current user. "
+                                "You may only have a single review resource.")
+
+    # If it wasn't healthy or running, we'll delete the existing resource if not already deleted, and then create a new one
+    logging.info("Existing review resource is in an unhealthy state.")
+    if existing_resource.deploymentStatus != "deleted":
+        logging.info("Deleting existing user resource...")
+        _ = await delete_review_user_resource(
+            user_resource=existing_resource,
+            user_resource_repo=user_resource_repo,
+            workspace_service_repo=workspace_service_repo,
+            resource_template_repo=resource_template_repo,
+            operations_repo=operation_repo,
+            resource_history_repo=resource_history_repo,
+            user=user
+        )
+
```

Comment:
this is the logic from the airlock route for `create_review_user_resource`,
broke it to multiple methods